packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,633 @@
+
+```ts
+...
+export class SelectorController<
+    State,
+    Action extends RAction,
+    Result = unknown
+> implements ReactiveController {
+  ...
+  private _unsubscribe: () => void;
+  ...
+
+  hostConnected() {
+    this._unsubscribe = this._store.subscribe(() => {
+      ...
+    });
+  }
+
+  hostDisconnected() {
+    this._unsubscribe();
+  }
+}
+```
+
+We now have a Redux `SelectorController` Reactive Controller that listens to a Redux store for state changes, selects a value from the state, and updates the host element whenever that state value changes!
+
+## Using the Controllers
+
+Now that we have a functioning controller, let’s use it! Let’s create two components and update our shape dial’s controls:
+
+1. `shape-count`
+    - A component that counts the shapes and the total number of shapes
+2. `shape-list`
+    - A component that visualizes the shapes that we have added
+
+And finally we will update `shape-dials` to disable the buttons when they are not applicable. The application should look something like this:
+
+<img
+    src="/images/articles/redux-reactive-controllers/application.webp"
+    loading="lazy"
+    width="550px"
+    title="Sketch of the redux application"
+    alt="A vector box diagram of the entire redux application. There are 3 sections. The first is the shape-dials component – it has 3 rows. The first row is a minus button, the label 'squares', followed by a plus button. The second row is a minus button, the label 'circles', followed by a plus button. The final row is a long 'reset' button. This component is surrounded by a red dashed-square labeled '<shape-dials>'. The next section to the right of the shape-dials section is labeled by a red label that says '<shape-count>'. It is a black box with 3 rows of text. The first row is '2 circles'. The second row is '2 squares', and the third row is '4 total'. Below the shape-dials and shape-count components is a long rectangle labeled '<shape-list>'. This is a long, horizonal, black rectangle and inside of it is a red square followed by a blue circle, blue circle, and another red square.">
+
+### shape-count
+
+`shape-count` should be a component that only subscribes and reads from the store. Let us create a custom element and render the table:
+
+```ts
+import { LitElement, html} from 'lit';
+import { customElement} from 'lit/decorators.js';
+
+@customElement('shape-count')
+export class ShapeCount extends LitElement {
+  render() {
+    return html`
+      <table>
+        <tr>
+          <td>${0}</td>
+          <td>circles</td>
+        </tr>
+        <tr>
+          <td>${0}</td>
+          <td>squares</td>
+        </tr>
+        <tr>
+          <td>${0}</td>
+          <td>total</td>
+        </tr>
+      </table>
+    `;
+  }
+}
+```
+
+Next we want to render actual counts rather than just `0`. In this case we will need all values from the state:
```

Comment:
Nit: `In this case we will need all values from the state:`.
Maybe:

`In this case we will need to pull those values from the store`.